config/utils.py

```diff
@@ -0,0 +1,160 @@
+        color = self.COLORS.get(record.levelno, self.RESET)
+        message = super().format(record)
+        return f"{color}{message}{self.RESET}"
+
+
+def parse_devices_output(file_path):
+    gpu_info = defaultdict(lambda: {})
+
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    for line in lines:
+        # Check for NVIDIA GPUs
+        if "NVIDIA" in line and "GPUs" not in line:
+            # Extract NVIDIA models using a simpler regex
+            match = re.search(r'NVIDIA (.+)', line)
+            if match:
+                model = match.group(1).strip()
+                if model not in gpu_info["NVIDIA"]:
+                    gpu_info["NVIDIA"].update({model : 1})
+                else:
+                    gpu_info["NVIDIA"][model] += 1
+
+        # Check for AMD GPUs
+        elif "AMD" in line or "Radeon" in line:
+            match = re.search(r'(\w+ \w+)', line)
+            if match:
+                model = match.group(0).strip()
+                if model not in gpu_info["AMD"]:
+                    gpu_info["AMD"].update({model : 1})
+                    print(gpu_info["AMD"])
+                else:
+                    gpu_info["AMD"][model] += 1
+
+        # Check for Intel GPUs
+        elif "Intel" in line:
+            match = re.search(r'(\w+ \w+)', line)
+            if match:
+                model = match.group(0).strip()
+                if model not in gpu_info["Intel"]:
+                    gpu_info["Intel"].update({model : 1})
+                    print(gpu_info["Intel"])
+                else:
+                    gpu_info["Intel"][model] += 1
+
+    return gpu_info
+
+
+def parse_containers_output(file_path : str):
+
+    containers_info = []
+    containers_found = False
+
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    for line in lines:
+        if "Installed containers" in line:
+            containers_found = True
+        elif containers_found:
+            type = line.split(' modules: ')[0].strip()
+            try:
+                modules = line.split(' modules: ')[1].split(', ')
+                modules = [m.strip() for m in modules]
+                modules.append(type)
+            except:
+                modules = []
+            containers_info.append({'type': type, 'modules': modules})
+
+    return containers_info
+
+
+def parse_cnmemory_output(file_path : str):
+
+    cnmemory_info = None
+    cnmemory_found = False
+
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    for line in lines:
+        if "Total memory:" in line:
+            match = re.search(r'(\d+)Gi', line)
+            if match:
+                cnmemory_info = match.group(1)
+                break
+        
+    return cnmemory_info
+
+
+def generate_submission_file(containers : bool, devices : bool, cn_memory : bool, partition_name : str, access : list = []):
+
+    with open("autodetection_job.sh", "w") as file:
```

Comment:
I don't like that curerently we leave the autodetction files where the script runs. Could you have a look if it would be possible to use a temporary directory (similar to what reframe does for autodetection) and just print it so that the user can have a look at the script/output in case of errors